docs/advanced/automated-testing.md

```diff
@@ -0,0 +1,262 @@
+---
+title: Unit Testing
+---
+
+## Getting started
+
+This guide is going to explore unit testing, it’s assuming some familiarity with the subject, but it’s not required.
+
+So to start off, you’re gonna need some project set up, the specifics don’t really matter, you can just take these ideas and extrapolate them to your own needs, but for demonstration I’m going to have a basic project set up like this:
+
+![Project setup](/images/testing-project-setup.png)
+
+Our code is very simple for this example.
+
+```typescript title=index.ts
+export const libraryFunction = (param: number[], reverse: boolean) => {
+  let reversed: number[] = [];
+
+  if (reverse) {
+    param.forEach((n) => table.insert(reversed, 1, n));
+  } else {
+    reversed = param;
+  }
+
+  return table.concat(reversed, ",");
+};
+```
+
+So our function takes an array of numbers like “[1,2,3,4]” and returns a string like “1,2,3,4” or “4,3,2,1”.
+
+This is where unit testing comes in, say a piece of code depends on this function working like it does, and for it to keep working exactly like it does right now, obviously you could use the tried and true method of doing the testing yourself, but unit testing provides an alternative to the manual work.
+
+So, unit testing, where do we begin?
+
+To start off we’re going to need these tools installed:
+
+https://luarocks.org/
+&
+https://olivinelabs.com/busted/
+
+_Luarocks is known to be hard to impossible to install on Windows systems sucessfully, if that’s the case for you, definitely keep reading on, we’re gonna explore a (free) option to do away with running tests locally entirely._
+
+Now that we have busted installed and the `busted` command runs in your terminal of choice successfully we can proceed.
+
+Let’s quickly install busted typings like so `npm install -d busted-tstl`
+
+And we need to add busted-tstl types to our tsconfig.json, you should already have a section like this:
```

Comment:
This is only required in the tsconfig in your test directory, also in your example repository I would suggest moving tsconfig.test.json to spec/tsconfig.json, so you are not polluting your main codebase with testing stuff.

Reply:
I would not assume people already have `types` in their tsconfig, and I think it would be a good idea to mention `compilerOptions`, like we do on [the language extensions page](https://typescripttolua.github.io/docs/advanced/language-extensions).